lib/models/question_catalog/question_catalog_reader.dart

```diff
@@ -3,27 +3,36 @@ import 'dart:convert';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
-
 import 'question_catalog.dart';
 
 class QuestionCatalogReader {
-  late Locale deviceLocale;
   final String defaultLocate;
 
   QuestionCatalogReader({this.defaultLocate = 'en'});
 
   Locale get _deviceLocale => PlatformDispatcher.instance.locale;
 
-  Future<QuestionCatalog> read() async {
+  Future<QuestionCatalog> read(bool advanceCatalog) async {
+    var questionCatalog = await _readCatalog('question_catalog');
+
+    if (advanceCatalog) {
+      final advanceQ = await _readCatalog('advanced_question_catalog');
+      questionCatalog.addAll(advanceQ);
```

Comment:
This will fail, since `QuestionCatalog` is immutable.

Maybe you can add a separate method called and combine the catalogs like this:

```dart
Future<QuestionCatalog> readAll(Iterable<String> assetPaths) {
final catalogs = await Future.wait(
  assetPaths.map((assetPath) => read(assetPath)),
);

return QuestionCatalog(
  catalogs.expand((list) => list).toList(),
);
}
```

Note: There is one problem though, the QuestionDefinitions get a unique runtime id which is currently derived from the indexes. When merging two catalogs there will be duplicated ids. So the above function actually needs to change to:


```dart
Future<QuestionCatalog> readAll(Iterable<String> assetPaths) {
final catalogs = await Future.wait(
  // use _readCatalog, because combining two catalogs e.g. via + operator will lead to duplicated `runtimeId`s for the `QuestionDefinition`s.
  assetPaths.map((assetPath) => _readCatalog(assetPath)),
);
return QuestionCatalog.fromJson(catalogs.expand((list) => list));
}
```
`QuestionCatalog.fromJson` needs to change to `QuestionCatalog.fromJson(Iterable<Map<String, dynamic>> json) :`

Reply:
I did the mentioned changes. However, I want to make a comment regarding "This will fail since QuestionCatalog is immutable."I tested the code and It does not present any problem. I switch between "with professional" and "without professional" (Taking into account closing and opening the app since it can not be updated in real-time) and it did the reading cataloged and appended it properly.

Reply:
My bad, I thought `_readCatalog` would return a `QuestionCatalog`, but since it returns a `List` it should work.